Setup/Model/ProductProtection/ProtectionPlanProduct20230714.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/*
+ * Copyright Extend (c) 2023. All rights reserved.
+ * See Extend-COPYING.txt for license details.
+ */
+
+namespace Extend\Integration\Setup\Model\ProductProtection;
+
+use Exception;
+use Extend\Integration\Service\Extend;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Catalog\Model\Product\Attribute\Source\Status;
+use Magento\Catalog\Model\Product\Type;
+use Magento\Catalog\Model\Product\Visibility;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\ProductFactory;
+use Magento\Eav\Api\Data\AttributeSetInterface;
+use Extend\Integration\Setup\Model\AttributeSetInstaller;
+use Magento\Store\Model\StoreManagerInterface;
+
+// NOTE: make sure to update ProductInstaller's use statement with this class if it is the current version
+class ProtectionPlanProduct20230714 implements ProtectionPlanProductInterface
+{
+    const VERSION = '2023-07-14';
+
+    private ProductFactory $productFactory;
+    private ProductRepositoryInterface $productRepository;
+    private AttributeSetInstaller $attributeSetInstaller;
+    private StoreManagerInterface $storeManager;
+
+    public function __construct(
+        ProductFactory $productFactory,
+        ProductRepositoryInterface $productRepository,
+        AttributeSetInstaller $attributeSetInstaller,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->productFactory = $productFactory;
+        $this->productRepository = $productRepository;
+        $this->attributeSetInstaller = $attributeSetInstaller;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * Create the protection plan product
+     *
+     * @return \Magento\Catalog\Model\Product
+     */
+    public function createProduct()
+    {
+        $product = $this->productFactory->create();
+        $attributeSet = $this->attributeSetInstaller->createAttributeSet();
```

Comment:
I would like to keep this separate, and pass in $attributeSet as we've been doing.  We shouldn't always assume both need to be created.  This model's purpose is to create the product, and that's model's purpose is to create the attribute set.  If both need to be create/recreated, they can both be called from the original call.